src/components/Categories.jsx

```diff
@@ -1,4 +1,17 @@
+import { useDispatch } from 'react-redux';
 import styles from '../styles/Categories.module.css';
+import { checkStatus } from '../redux/categories/categoriesSlice';
 
-const Categories = () => <button type="button" className={styles.checkBtn}>Check status</button>;
+const Categories = () => {
+  const dispatch = useDispatch();
+  return (
+    <button
+      type="button"
+      className={styles.checkBtn}
+      onClick={() => dispatch(checkStatus())}
+    >
+      Check status
```

Comment:
Kindly ensure that the 'Under Construction' status gets rendered when the check status button is clicked. You currently don't seem to have a component to display the status.

Reply:
Dear @emmiiorji, thanks for review. I checked the rendering by log something and then I eased it. but I didn't see in the requirements that we need to display something, I thought it says I need to only add a functionality. so should I display something near the button?

Reply:
Hi @mohisa302

Thank you for your response. You have well implemented the check status action and it works. However, the state isn't used in the categories component just as the books state is used within the books component.

Reply:
Thanks @emmiiorji , I've already added in a button, so would you explain how can I use it? because in the requirements I haven't seen anything about what functionality sould I add in category

Reply:
Here's an example of how you can use the state within the component:

```javascript
const dispatch = useDispatch();
const status = useSelector((state) => state.categories);
  return (
   <>
      <button
        type="button"
        className={styles.checkBtn}
        onClick={() => dispatch(checkStatus())}
      >
        Check status
      </button>
      <h1 className="status">{status}</h1>
    </>
```

Reply:
Thanks @emmiiorji for your explanation I got it